### DISCUSSION TITLE
Best way to implement the drawing functionality on react-map-gl

### DISCUSSION THREAD

**Discussion by SuhaibMaraqa** [uber/react-map-gl#2382] [General]
As I get there is a package tailored for react-map-gl called [react-map-gl-draw](https://www.npmjs.com/package/react-map-gl-draw), I don't think it's updated to the last version of react-map-gl and the documentation is complicated. Also there is the draw polygon [example](https://github.com/visgl/react-map-gl/tree/7.0-release/examples/draw-polygon) that react-map-gl implemented using @mapbox/mapbox-gl-draw package. So which one to choose to start implementing the MapboxDraw functionality with react-map-gl now ?

**Comment by Pessimistress** [comment]
react-map-gl-draw was made for react-map-gl v5 and v6.

As of v7.0 react-map-gl supports Mapbox/Maplibre plugins.

Generally speaking, the examples reflect the latest recommendation.
